records_mover/records/delimited/sniff.py

```diff
@@ -119,15 +59,15 @@ def sniff_hints_from_fileobjs(fileobjs: List[IO[bytes]],
 
 
 def infer_newline_format(fileobj: IO[bytes],
-                         inferred_hints: RecordsHints,
+                         inferred_hints: MutableRecordsHints,
                          encoding_hint: str) -> None:
     closed = False
     if getattr(fileobj, 'closed', None) is not None:
         closed = fileobj.closed
     if closed or not fileobj.seekable():
         logger.warning("Assuming UNIX newline format, as stream is not rewindable")
         return
-    python_encoding = python_encoding_from_hint[encoding_hint]
+    python_encoding = python_encoding_from_hint[encoding_hint]  # type: ignore
```

Comment:
I've got another PR outstanding to improve hint stiffing, so I didn't go too far to improve the use of types here yet.